selfdrive/car/tests/test_fw_fingerprint.py

```diff
@@ -134,13 +137,27 @@ def _benchmark(brand, num_pandas, n):
 
     return round(sum(times) / len(times), 2)
 
-  def _assert_timing(self, avg_time, ref_time, tol):
-    self.assertLess(avg_time, ref_time + tol)
-    self.assertGreater(avg_time, ref_time - tol, "Performance seems to have improved, update test refs.")
+  def _assert_timing(self, avg_time, ref_time):
+    self.assertLess(avg_time, ref_time + self.TOLERANCE)
+    self.assertGreater(avg_time, ref_time - self.TOLERANCE, "Performance seems to have improved, update test refs.")
 
   def test_fw_query_timing(self):
-    tol = 0.1
-    total_ref_time = 4.6
+    """
+    Tests timing if full timeout time is taken for:
+      - VIN query
+      - Present ECU queries
+      - FW queries for each brand
+      - Total time taken for a car which doesn't match to any brand (CAN fp)
+    """
+
+    # Includes all FW queries + queries in functions
+    total_ref_time = 6.2
+    func_ref_times = {
+      'get_vin': 1.05,
+      'get_present_ecus': 0.5,
```

Comment:
realized we can group the first parallel Toyota address query into the parallel queries to reduce the fingerprinting time by 0.1s for all cars, and 0.2 for Toyota, added to https://github.com/commaai/openpilot/issues/27665